### PR TITLE
Feature/user update functionality

### PR DESCRIPTION
**Pull request by HarisShahidHameed**
Trello Link : https://trello.com/c/iBOnOcgh/10-functionality-to-update-user

### CODE REVIEW

File: server/libs/userlibs.js [hunk 1]
```diff
@@ -18,11 +19,12 @@ exports.register = function(req,res) {
 //login functionality for the app. Checks whether user exist and if exist the information provided is correct
 exports.login = function(req,res){
     let body = _.pick(req.body , ['email','password']);
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: we don't need that you can get body.email and body.password without that

File: server/libs/userlibs.js [hunk 2]
```diff
@@ -18,11 +19,12 @@ exports.register = function(req,res) {
 //login functionality for the app. Checks whether user exist and if exist the information provided is correct
 exports.login = function(req,res){
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: space  before ( and after )

File: server/libs/userlibs.js [hunk 3]
```diff
@@ -18,11 +19,12 @@ exports.register = function(req,res) {
...
     let body = _.pick(req.body , ['email','password']);
+    let data = {};
 
     const validateUser = function(email,password){
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: same as above

File: server/libs/userlibs.js [hunk 4]
```diff
@@ -18,11 +19,12 @@ exports.register = function(req,res) {
...
     return User.findOne({email}).then((user) => {
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: code is not indent

File: server/libs/userlibs.js [hunk 5]
```diff
@@ -33,7 +35,7 @@ exports.login = function(req,res){
                 if(res){
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: space after )

File: server/libs/userlibs.js [hunk 6]
```diff
@@ -33,7 +35,7 @@ exports.login = function(req,res){
...
                     resolve(user);
                 }else{
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: spacing } else {

File: server/libs/userlibs.js [hunk 7]
```diff
@@ -48,4 +50,64 @@ exports.login = function(req,res){
     }).catch((error) => {
         res.status(error.status).send(error.message);
     });
+};
+
+exports.updateUser = function(req,res) {
+    let body = req.body;
+
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: remove spacing

File: server/libs/userlibs.js [hunk 8]
```diff
@@ -48,4 +50,64 @@ exports.login = function(req,res){
...
+    const validationForNameExistence = (object) => {
+        if("name" in object){
+            return true;
+        }else{//eslint-disable-line
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: why disabling eslint. shouldn't disable eslint without discussion or permission

Comment (abdulhaqshah) [inline server/libs/userlibs.js]: spacing

File: server/libs/userlibs.js [hunk 9]
```diff
@@ -48,4 +50,64 @@ exports.login = function(req,res){
...
+            return false;
+        }
+    }
+
+    const validationForEmailExistence = (object) => {
+        if("email" in object){
+            return true;
+        }else{//eslint-disable-line
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: same and apply on all the undiscussed disability

File: server/libs/userlibs.js [hunk 10]
```diff
@@ -48,4 +50,64 @@ exports.login = function(req,res){
...
+            return false;
+        }
+    }
+
+    const emptyCheck = (object) => {
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: why too much validation function

File: server/libs/userlibs.js [hunk 11]
```diff
@@ -48,4 +50,64 @@ exports.login = function(req,res){
...
+        if(object.length === 0){
+            return false;
+        }else{//eslint-disable-line
+            return true;
+        }
+    }
+
+    const update = (object) => {
+        User.findByIdAndUpdate(object.id, object, {new: true}, (err, doc) => {
+            if(doc){
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: same spacing issue will apply on all function and if in this PR

File: server/utilities/constants.js [hunk 12]
```diff
@@ -11,7 +11,10 @@ const messages = {
     created : "has been created successfully",
     bad_request : "This is is a bad request",
     not_match : "does not match",
-    not_found : "has not been found"
+    not_found : "has not been found",
```
Comment (abdulhaqshah) [inline server/utilities/constants.js]: should be indent like  : will be in center

**Review by abdulhaqshah — CHANGES_REQUESTED** [review]

File: server/libs/userlibs.js (hunk 2 above)
Comment (HarisShahidHameed) [reply server/libs/userlibs.js]: Done

File: server/libs/userlibs.js (hunk 1 above)
Comment (HarisShahidHameed) [reply server/libs/userlibs.js]: Removed that.

File: server/libs/userlibs.js (hunk 3 above)
Comment (HarisShahidHameed) [reply server/libs/userlibs.js]: Done.

File: server/libs/userlibs.js (hunk 4 above)
Comment (HarisShahidHameed) [reply server/libs/userlibs.js]: Done.

File: server/libs/userlibs.js (hunk 5 above)
Comment (HarisShahidHameed) [reply server/libs/userlibs.js]: Done.

File: server/libs/userlibs.js (hunk 6 above)
Comment (HarisShahidHameed) [reply server/libs/userlibs.js]: Done

File: server/libs/userlibs.js (hunk 9 above)
Comment (HarisShahidHameed) [reply server/libs/userlibs.js]: Done.

File: server/libs/userlibs.js (hunk 10 above)
Comment (HarisShahidHameed) [reply server/libs/userlibs.js]: Made it into one.

File: server/utilities/constants.js (hunk 12 above)
Comment (HarisShahidHameed) [reply server/utilities/constants.js]: Done.

File: server/libs/userlibs.js [hunk 13]
```diff
@@ -1,113 +1,116 @@
...
+        case 2:
             emailValidation(body);
-        }else{
-            res.status(statusCodes.forbidden).send(`Name ${messages.empty}`);
-        }
-    }else{
-        emailValidation(body);
+            break;
+        case 3:
+            if (emptyCheck(body.name)) {
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: use conditional operator for checking one line : body.name ? true : false  this is example

File: server/libs/userlibs.js [hunk 14]
```diff
@@ -1,113 +1,116 @@
...
+                update(body);
+            }
+            break;
+        case 4:
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: use proper naming convention

File: server/libs/userlibs.js [hunk 15]
```diff
@@ -1,113 +1,116 @@
...
+        if (object.length === 0) {
             return false;
-        }else{//eslint-disable-line
-            return true;
         }
+            return true;
+        
     }
 
     const update = (object) => {
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: user proper name which define what the function is doing

File: server/libs/userlibs.js [hunk 16]
```diff
@@ -1,51 +1,99 @@
...
+            if (doc) {
+                res.status(statusCodes.successful).send({message : `User ${messages.updated}`, data : doc})
+            } else {
+                res.status(statusCodes.not_found).send(`User ${messages.not_found}`);
+            }
+        })
+    }
+
+    const validateEmailAndUpdate = (object) => {
+            if(validator.isEmail(object.email)){
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: This is not indent sir

**Review by abdulhaqshah — APPROVED** [review]

**Pull request merged**